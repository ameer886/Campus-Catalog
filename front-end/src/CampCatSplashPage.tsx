import React from 'react';
import './CampCatSplashPage.css';

const CampCatSplashPage: React.FunctionComponent = () => {
  return (
    <div className="App">
      <h1>Hello world!</h1>
    </div>
  );
};

export default CampCatSplashPage;
