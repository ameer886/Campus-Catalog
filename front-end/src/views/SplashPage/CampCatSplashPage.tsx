import React from 'react';
import './CampCatSplashPage.css';

import GenericTable, {
  ColumnDefinitionType,
} from '../../components/GenericTable/GenericTable';

/*
 * The Splash Page
 * This is where everyone will arrive by default
 * Should contain links to all model pages and the about page
 */
const CampCatSplashPage: React.FunctionComponent = () => {
  type IceCream = {
    flavor: string;
    price: number;
  };

  const exampleRows: Array<IceCream> = [
    {
      flavor: 'Vanilla',
      price: 4.31,
    },
    {
      flavor: 'Chocolate',
      price: 5.88,
    },
    {
      flavor: 'Neopolitan',
      price: 6.99,
    },
  ];
  const exampleHeaders: ColumnDefinitionType<
    IceCream,
    keyof IceCream
  >[] = [
    {
      key: 'flavor',
      header: 'Flavor',
      sortFunc: (a: string, b: string) => a.localeCompare(b),
    },
    {
      key: 'price',
      header: 'Price',
      sortFunc: (a: number, b: number) => a - b,
    },
  ];

  return (
    <div className="App">
      <h1>Hello world!</h1>
      <GenericTable
        columnDefinitions={exampleHeaders}
        data={exampleRows}
      />
    </div>
  );
};

export default CampCatSplashPage;
