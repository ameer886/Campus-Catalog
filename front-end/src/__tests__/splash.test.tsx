import React from 'react';
import { render, screen } from '@testing-library/react';

import CampCatSplashPage from '../views/SplashPage/CampCatSplashPage';

describe('Splash Page Test Suite', () => {
  it('page displays title', () => {
    render(<CampCatSplashPage />);
    const textElement = screen.getByText(/Campus Catalog/i);
    expect(textElement).toBeInTheDocument();
  });
});
