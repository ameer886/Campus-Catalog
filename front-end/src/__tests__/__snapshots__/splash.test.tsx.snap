// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash Page Test Suite snapshot test 1`] = `ShallowWrapper {}`;
